src/go/intelligence/pipeline/silence_alert.go

```diff
@@ -0,0 +1,128 @@
+package pipeline
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"go/intelligence/models"
+	"go/intelligence/utils"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"time"
+)
+
+// Module     : intelligence
+// Author     : Rahul Indra <indrarahul2013 AT gmail dot com>
+// Created    : Wed, 1 July 2020 11:04:01 GMT
+// Description: CERN MONIT infrastructure Intelligence Module
+
+//Silence function silences the old alert
+func Silence(data <-chan models.AmJSON) <-chan models.AmJSON {
+
+	silencedData := make(chan models.AmJSON)
+	go func() {
+		for each := range data {
+			err := silenceAlert(each)
+			if err != nil {
+				log.Printf("Could not silence alert, error:%v\n", err)
+				if utils.ConfigJSON.Verbose > 1 {
+					log.Printf("Silence Data: %s\n ", each)
+				}
+			}
+			silencedData <- each
+		}
+		close(silencedData)
+	}()
+	return silencedData
+}
+
+func silenceAlert(data models.AmJSON) error {
+
+	apiurl := utils.ValidateURL(utils.ConfigJSON.CMSMONURL, utils.ConfigJSON.PostSilenceAPI) // POST API for creating silences.
+
+	var sData models.SilenceData
+	var alertnameMatcher models.Matchers
+	var severityMatcher models.Matchers
+
+	sData.StartsAt = data.StartsAt //Start Time equal to main alert	   //So that the silence remains on old alert till the lifetime of the alert
+	sData.EndsAt = data.EndsAt     //End Time equal to main alert	  //So that the silence remains on old alert till the lifetime of the alert
+	sData.CreatedBy = utils.ConfigJSON.CreatedBy
+	sData.Comment = utils.ConfigJSON.Comment
+
+	alertnameMatcher.Name = utils.ConfigJSON.UniqueLabel
+	severityMatcher.Name = utils.ConfigJSON.SeverityLabel
+
+	for k, v := range data.Labels {
+		if k == utils.ConfigJSON.UniqueLabel {
+			if val, ok := v.(string); ok {
+				alertnameMatcher.Value = val
+			}
+		}
+
+		if k == utils.ConfigJSON.SeverityLabel {
+			if data.Labels["service"] == "SSB" {
```

Comment:
This should be generalized via config map of severity levels.

Reply:
Fixed this.